tbx/work/models.py

```diff
@@ -143,12 +157,131 @@ def type(self):
+    related_services = ParentalManyToManyField(
+        "taxonomy.Service", related_name="case_studies"
+    )
+
+    content_panels = Page.content_panels + [
+        FieldPanel("intro"),
+        InlinePanel("authors", label="Author", min_num=1),
+        FieldPanel("client", classname="client"),
+        FieldPanel("date"),
+        MultiFieldPanel(
+            [
+                FieldPanel("header_image"),
+                FieldPanel("header_caption"),
+                FieldPanel("header_attribution"),
+            ],
+            heading="Header image",
+        ),
+        FieldPanel("body"),
+    ]
+
+    promote_panels = [
+        MultiFieldPanel(Page.promote_panels, "Common page configuration"),
+        FieldPanel("listing_summary"),
+        FieldPanel("related_services", widget=forms.CheckboxSelectMultiple),
+        MultiFieldPanel(SocialFields.promote_panels, "Social fields"),
+    ]
+
+    @property
+    def type(self):
+        return "CASE STUDY"
+
+    @cached_property
+    def services(self):
+        return self.related_services.all()
+
+    @property
+    def first_author(self):
+        """Safely return the first author if one exists."""
+        author = self.authors.first()
+        if author:
+            return author.author
+        return None
+
+    @property
+    def work_index(self):
+        ancestor = WorkIndexPage.objects.ancestor_of(self).order_by("-depth").first()
+
+        if ancestor:
+            return ancestor
+        else:
+            # No ancestors are work indexes,
+            # just return first work index in database
+            return WorkIndexPage.objects.live().public().first()
+
+    @property
+    def related_works(self):
+        return [
+            {
+                "client": case_study.client,
+                "title": case_study.title,
+                "url": case_study.url,
+                "author": case_study.first_author,
+                "date": case_study.date,
+                "read_time": case_study.read_time,
+                "related_services": case_study.related_services.all(),
+                "listing_image": case_study.header_image,
+            }
+            # get 3 pages with same services and exclude self page
+            for case_study in WorkPage.objects.filter(
+                related_services__in=self.services
+            )
+            .live()
+            .prefetch_related("related_services")
+            .defer_streamfields()
+            .distinct()
+            .order_by("-id")
+            .exclude(pk=self.pk)[:3]
+        ]
+
+    def set_body_word_count(self):
+        body_basic_html = self.body.stream_block.render_basic(self.body)
+        body_text = BeautifulSoup(body_basic_html, "html.parser").get_text()
+        remove_chars = string.punctuation + "“”’"
+        body_words = body_text.translate(
+            body_text.maketrans(dict.fromkeys(remove_chars))
+        ).split()
+        self.body_word_count = len(body_words)
```

Comment:
Is this a pre-existing pattern copied from somewhere else?

Reply:
Yes this is what's been used on the previous build. It's also used on the `BlogPage`